core/src/SYCL/Kokkos_SYCL_ParallelReduce_Range.hpp

```diff
@@ -318,19 +326,32 @@ class Kokkos::Impl::ParallelReduce<CombinedFunctorReducerType,
         cgh.parallel_for(
             sycl::nd_range<1>(n_wgroups * wgroup_size, wgroup_size),
             reduction_lambda);
-      });
+      };
+
+#ifdef SYCL_EXT_ONEAPI_GRAPH
+      if constexpr (Policy::is_graph_kernel::value) {
+        sycl_attach_kernel_to_node(*this, cgh_lambda);
+      } else
+#endif
+      {
+        last_reduction_event = q.submit(cgh_lambda);
 #ifndef KOKKOS_IMPL_SYCL_USE_IN_ORDER_QUEUES
-      q.ext_oneapi_submit_barrier(
-          std::vector<sycl::event>{parallel_reduce_event});
+        q.ext_oneapi_submit_barrier(
+            std::vector<sycl::event>{last_reduction_event});
 #endif
-      last_reduction_event = parallel_reduce_event;
+      }
     }
 
     // At this point, the reduced value is written to the entry in results_ptr
     // and all that is left is to copy it back to the given result pointer if
     // necessary.
     // Using DeepCopy instead of fence+memcpy turned out to be up to 2x slower.
     if (host_result_ptr) {
+      if constexpr (Policy::is_graph_kernel::value)
+        Kokkos::abort(
+            "parallel_reduce not implemented for graph kernels if result is "
+            "not device-accessible!");
```

Comment:
Is that something we could turn into a compile error?
(Not necessarily blocking, just wondering if you look into this and if it is doable)

Reply:
It's a little tricky since we don't know at compile-time at this point whether parallel_reduce was called with a result parameter or not.